cairo/ethereum/cancun/vm/instructions/system.cairo

```diff
@@ -1,58 +1,253 @@
+            code=code,
+            depth=Uint(evm.value.message.value.depth.value + 1),
+            should_transfer_value=should_transfer_value,
+            is_static=is_static,
+            accessed_addresses=child_accessed_addresses,
+            accessed_storage_keys=child_accessed_storage_keys,
+            parent_evm=evm,
+        ),
+    );
+
+    // prepare arguments to jump to process_message
+    [ap] = range_check_ptr, ap++;
+    [ap] = bitwise_ptr, ap++;
+    [ap] = keccak_ptr, ap++;
+    [ap] = poseidon_ptr, ap++;
+    [ap] = child_message.value, ap++;
+    [ap] = env.value, ap++;
+
+    call abs process_message_label;
+
+    let range_check_ptr = [ap - 6];
+    let bitwise_ptr = cast([ap - 5], BitwiseBuiltin*);
+    let keccak_ptr = cast([ap - 4], KeccakBuiltin*);
+    let poseidon_ptr = cast([ap - 3], PoseidonBuiltin*);
+    let child_evm_ = cast([ap - 2], EvmStruct*);
+    let err = cast([ap - 1], EthereumException*);
+    tempvar child_evm = Evm(child_evm_);
+
+    if (cast(err, felt) != 0) {
+        // TODO: <https://github.com/kkrt-labs/keth/issues/568> we still need to drop dicts and get the child evm state etc
+        return err;
+    }
+
+    // The previous operations have mutated the `env` passed to the function, which
+    // is now located in child_evm.value.env. Notably, we must rebind env.state and env.transient_storage
+    // to their new mutated values. The reset is handled on a per-case basis in incorporate_child.
+    let updated_state = child_evm.value.env.value.state;
+    let updated_transient_storage = child_evm.value.env.value.transient_storage;
+    let old_env = evm.value.env;
+    EnvImpl.set_state{env=old_env}(updated_state);
+    EnvImpl.set_transient_storage{env=old_env}(updated_transient_storage);
+    EvmImpl.set_env{evm=evm}(old_env);
+
+    if (cast(child_evm.value.error, felt) != 0) {
+        incorporate_child_on_error(child_evm);
+        EvmImpl.set_return_data(child_evm.value.output);
+        let stack = evm.value.stack;
+        push{stack=stack}(U256(new U256Struct(0, 0)));
+        EvmImpl.set_stack(stack);
+
+        tempvar evm = evm;
+        tempvar poseidon_ptr = poseidon_ptr;
+        tempvar keccak_ptr = keccak_ptr;
+        tempvar bitwise_ptr = bitwise_ptr;
+        tempvar range_check_ptr = range_check_ptr;
+    } else {
+        incorporate_child_on_success(child_evm);
+        EvmImpl.set_return_data(child_evm.value.output);
+        let stack = evm.value.stack;
+        push{stack=stack}(U256(new U256Struct(1, 0)));
+        EvmImpl.set_stack(stack);
+
+        tempvar evm = evm;
+        tempvar poseidon_ptr = poseidon_ptr;
+        tempvar keccak_ptr = keccak_ptr;
+        tempvar bitwise_ptr = bitwise_ptr;
+        tempvar range_check_ptr = range_check_ptr;
+    }
+    let evm = evm;
+    let poseidon_ptr = poseidon_ptr;
+    let keccak_ptr = keccak_ptr;
+    let bitwise_ptr = bitwise_ptr;
+    let range_check_ptr = range_check_ptr;
```

Comment:
is this required?

Reply:
if I delete it I get reference revoked errors